### PR TITLE
Initial set of tests for the type.1 rule.

### PR DESCRIPTION
**Pull request by sergiyor**
This also includes parse_expectations.cpp which describes the format used to specify expectations.

### CODE REVIEW

**Comment by neilmacintosh** [comment]
@sergiyor So the idea here is to propose a set of test cases for the enforcement of each rule and a simple format for describing the results that are expected from a tool that wants to enforce that rule. At least, for each rule that is deterministically enforceable.

Am I right?

**Comment by sergiyor** [comment]
@neilmacintosh Yes, you are right.

File: tests/parse_expectations.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#include <string>
+#include <iostream>
+
+void main()
+{
```
Comment (gdr-at-ms) [inline tests/parse_expectations.cpp]: should be `int main()`

**Comment by neilmacintosh** [comment]
@mgehre would you have time to take a look at this PR and see if is something that clang-tidy developers working on supporting the core guidelines might also find useful? Feel free to loop in anyone else you think would be relevant.

**Comment by AndrewPardoe** [comment]
@jwakely: Could you please review this as well?

**Comment by neilmacintosh** [comment]
@jwakely we would love any GCC-informed feedback on the suggested approach here as well.

**Comment by neilmacintosh** [comment]
We are going to create a separate repository for these tests and the `parse_expectations` tool. These tests are currently used to validate CppCoreCheck, so it is useful to ensure they are recognized as separate from the Guidelines. However, any other tool vendors are welcome to join us in using them and contributing to them.

**Pull request closed without merging**